Sources/Filters/Core/CleanPolyData/index.js

```diff
@@ -0,0 +1,485 @@
+import macro from 'vtk.js/Sources/macros';
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+import vtkMergePoints from 'vtk.js/Sources/Common/DataModel/MergePoints';
+import vtkPointLocator from 'vtk.js/Sources/Common/DataModel/PointLocator';
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData/index';
+import { DesiredOutputPrecision } from 'vtk.js/Sources/Common/DataModel/DataSetAttributes/Constants';
+import { VtkDataTypes } from 'vtk.js/Sources/Common/Core/DataArray/Constants';
+
+// ----------------------------------------------------------------------------
+// vtkCleanPolyData methods
+// ----------------------------------------------------------------------------
+
+function vtkCleanPolyData(publicAPI, model) {
+  // Set our classname
+  model.classHierarchy.push('vtkCleanPolyData');
+
+  // Point processing
+  function processPoint(
+    ptId,
+    inPts,
+    newPts,
+    inputPD,
+    outputPD,
+    pointMap,
+    numUsedPts
+  ) {
+    const newx = [0, 0, 0];
+
+    const x = inPts.getPoint(ptId);
```

Comment:
it would be better to not allocate a new array each time.
```
let tempX = [];
function processPoint(...) {
  inPts.getPoint(ptId, tempX);
  ...
}
```